app/components/time_based_activity_heatmap/time_based_activity_heatmap.js

```diff
@@ -0,0 +1,93 @@
+import { Controller } from '@hotwired/stimulus';
+
+export default class extends Controller {
+  static targets = ['heatmap'];
+
+  connect() {
+    this.fetchTimeBasedActivityData();
+  }
+
+  async fetchTimeBasedActivityData() {
+    const response = await fetch('charts/time-based-activity');
```

Comment:
why did you choose to use the javascript apexcharts package instead of the apexcharts gem that is also included? The gem also offers a [heatmap](https://github.com/styd/apexcharts.rb#heatmap-chart) which would probably reduce complexity.

Reply:
See answer :point_up: